### ISSUE TITLE
Arduino to Arduino 1-Wire protocol

### ISSUE THREAD

**Issue by ONajj92** [mikaelpatel/Arduino-OWI#10]
First of all, let me thank you for your amazing work on this project.

Currently, I am working on a project that use position sensors, each sensor should have its own microcontroller. The only constraint that the "Master" microcontroller only has one digital pin available that supports interrupt therefore I thought of using 1-wire as a communication protocol for the system.

I tried to use the provided master/slave libraries between an Arduino mega as a master and a nano as a slave, both are connected via a digital pin on  pulled up to the mega 5V pin through a 10K resistor.
 
I used the provided examples (.ino) files. Initially I was successful sending the number (120) to the nano, but after a while the code stopped working on the master side and continuously return an assert error (last!=owi.ERROR).

Do you have any idea what is going on?

Thank you in advance for your help.

**Comment by ONajj92** [comment]
Update: I have a theory: it seems that when I tested the Master and Slave templates I ran into the mentioned error because the Master was Scanning/Assigning address to the Slave but at that moment the Slave was sending data without waiting for the Master reset signal. I want to rest() the bus and make the slave wait for the rest signal before it starts transmitting. I am using this command  Slave::OWI<BOARD::D2>::reset() but apparently the method is expectinag an Object of type pin_t PIN which is D2 here based on the GPIO lib but I cannot figure out how to satisfy the function. 

Error message : 
ArduinoS:32:30: error: cannot call member function 'bool Slave::OWI<PIN>::reset() [with BOARD::pin_t PIN = (BOARD::pin_t)658u]' without object
